pkg/metrics/metrics.jsx

```diff
@@ -917,163 +918,141 @@ class MetricsHour extends React.Component {
 const invalidService = proxy => proxy.state === null;
 const runningService = proxy => ['running', 'starting'].indexOf(proxy.state) >= 0;
 
-const PCPConfig = ({ buttonVariant, firewalldRequest, needsLogout, setNeedsLogout }) => {
-    const [dialogVisible, setDialogVisible] = useState(false);
+const wait_cond = (cond, objects) => {
+    return new Promise((resolve, reject) => {
+        const check = () => {
+            if (cond()) {
+                objects.forEach(o => o.removeEventListener("changed", check));
+                resolve();
+            }
+        };
+        objects.forEach(o => o.addEventListener("changed", check));
+        check();
+    });
+};
+
+const PCPConfigDialog = ({
+    firewalldRequest,
+    needsLogout, setNeedsLogout,
+    s_pmlogger, s_pmproxy, s_redis, s_redis_server
+}) => {
+    const Dialogs = useDialogs();
+    const dialogInitialProxyValue = runningService(s_pmproxy) && (runningService(s_redis) || runningService(s_redis_server));
     const [dialogError, setDialogError] = useState(null);
-    const [dialogLoggerValue, setDialogLoggerValue] = useState(false);
-    const [dialogProxyValue, setDialogProxyValue] = useState(null);
-    const [dialogInitialProxyValue, setDialogInitialProxyValue] = useState(null);
+    const [dialogLoggerValue, setDialogLoggerValue] = useState(runningService(s_pmlogger));
+    const [dialogProxyValue, setDialogProxyValue] = useState(dialogInitialProxyValue);
```

Comment:
The service proxies are only used to initialize  the state of the dialog.  `runningService(s_pmlogger)` is called on every render, but the value is ignored on all but the first one.  That's how `useState` works.  It is a bit wasteful, and that would an argument to compute those values once when instantiating the dialog.

However, `dialogInitialProxyValue` is recomputed on every render, and is used when enabling/disabling. This is argueably correct: if the service gets enabled in the background, but remains disabled in the dialog, it will be disabled when clicking "Save".

Reply:
> We can't let the button show both actual and desired state.

Yes, on/off buttons, checkboxes, or other inputs that the user interacts directly with should not follow system state while the dialog is open.

However, we might want it to follow system state for lists that the user picks from, such as the zones defined for the firewall, or the list of devices to create a volume group from. The user doesn't edit the list itself in the dialog, so there is no conflict. But it might be useful to keep the list up-to-date so that the user doesn't need to close and re-open the just to refresh it.

Reply:
Thanks for the details. This is finnicky, and AFAICS this PR doesn't change the current behaviour, so let's not block on this.